libs/ui-components/src/components/form/RichValidationTextField.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+import {
+  Button,
+  FormGroup,
+  HelperText,
+  HelperTextItem,
+  HelperTextItemProps,
+  InputGroup,
+  InputGroupItem,
+  Popover,
+  PopoverPosition,
+  TextInput,
+  TextInputProps,
+} from '@patternfly/react-core';
+import { TimesIcon } from '@patternfly/react-icons/dist/js/icons/times-icon';
+import { CheckIcon } from '@patternfly/react-icons/dist/js/icons/check-icon';
+import { CheckCircleIcon } from '@patternfly/react-icons/dist/js/icons/check-circle-icon';
+import { ExclamationCircleIcon } from '@patternfly/react-icons/dist/js/icons/exclamation-circle-icon';
+import { InfoCircleIcon } from '@patternfly/react-icons/dist/js/icons/info-circle-icon';
+
+import successColor from '@patternfly/react-tokens/dist/js/global_palette_green_500';
+import dangerColor from '@patternfly/react-tokens/dist/js/global_danger_color_100';
+import infoColor from '@patternfly/react-tokens/dist/js/global_palette_blue_300';
+import { useField } from 'formik';
+
+interface RichValidationStatusProps {
+  validations: RichValidationTextFieldProps['validations'];
+  metaError?: string | Record<string, string>;
+  isRequired: boolean;
+  hasValue: boolean;
+}
+const RichValidationStatus = ({ isRequired, hasValue, validations, metaError }: RichValidationStatusProps) => {
+  const showRequiredError = isRequired && !hasValue;
+  return (
+    <HelperText component="ul" className="rich-input__rules">
```

Comment:
where is this className from ?

Reply:
I removed the classes since they were only being used in the Cypress tests but they were not strictly needed.